src/platform_impl/android/keycodes.rs

```diff
@@ -156,324 +159,393 @@ pub fn to_physical_keycode(keycode: Keycode) -> KeyCode {
-        TvTerrestrialAnalog => Key::TVTerrestrialAnalog,
-        TvTerrestrialDigital => Key::TVTerrestrialDigital,
-        TvSatellite => Key::TVSatellite,
-        TvSatelliteBs => Key::TVSatelliteBS,
-        TvSatelliteCs => Key::TVSatelliteCS,
-        TvSatelliteService => Key::TVSatelliteToggle,
-        TvNetwork => Key::TVNetwork,
-        TvAntennaCable => Key::TVAntennaCable,
-        TvInputHdmi1 => Key::TVInputHDMI1,
-        TvInputHdmi2 => Key::TVInputHDMI2,
-        TvInputHdmi3 => Key::TVInputHDMI3,
-        TvInputHdmi4 => Key::TVInputHDMI4,
-        TvInputComposite1 => Key::TVInputComposite1,
-        TvInputComposite2 => Key::TVInputComposite2,
-        TvInputComponent1 => Key::TVInputComponent1,
-        TvInputComponent2 => Key::TVInputComponent2,
-        TvInputVga1 => Key::TVInputVGA1,
-        TvAudioDescription => Key::TVAudioDescription,
-        TvAudioDescriptionMixUp => Key::TVAudioDescriptionMixUp,
-        TvAudioDescriptionMixDown => Key::TVAudioDescriptionMixDown,
-        TvZoomMode => Key::ZoomToggle,
-        TvContentsMenu => Key::TVContentsMenu,
-        TvMediaContextMenu => Key::TVMediaContext,
-        TvTimerProgramming => Key::TVTimer,
-        Help => Key::Help,
-        NavigatePrevious => Key::NavigatePrevious,
-        NavigateNext => Key::NavigateNext,
-        NavigateIn => Key::NavigateIn,
-        NavigateOut => Key::NavigateOut,
-        StemPrimary => Key::Unidentified(native),
-        Stem1 => Key::Unidentified(native),
-        Stem2 => Key::Unidentified(native),
-        Stem3 => Key::Unidentified(native),
-        DpadUpLeft => Key::Unidentified(native),
-        DpadDownLeft => Key::Unidentified(native),
-        DpadUpRight => Key::Unidentified(native),
-        DpadDownRight => Key::Unidentified(native),
-        MediaSkipForward => Key::MediaSkipForward,
-        MediaSkipBackward => Key::MediaSkipBackward,
-        MediaStepForward => Key::MediaStepForward,
-        MediaStepBackward => Key::MediaStepBackward,
-        SoftSleep => Key::Unidentified(native),
-        Cut => Key::Cut,
-        Copy => Key::Copy,
-        Paste => Key::Paste,
-        SystemNavigationUp => Key::Unidentified(native),
-        SystemNavigationDown => Key::Unidentified(native),
-        SystemNavigationLeft => Key::Unidentified(native),
-        SystemNavigationRight => Key::Unidentified(native),
-        AllApps => Key::Unidentified(native),
-        Refresh => Key::BrowserRefresh,
-        ThumbsUp => Key::Unidentified(native),
-        ThumbsDown => Key::Unidentified(native),
-        ProfileSwitch => Key::Unidentified(native),
+    let native = NativeKey::Android(keycode.into());
+
+    match key_char {
+        Some(KeyMapChar::Unicode(c)) => {
+            Key::Character(smol_str::SmolStr::from_iter([c].into_iter()))
+        }
+        Some(KeyMapChar::CombiningAccent(c)) => Key::Dead(Some(c)),
+        None | Some(KeyMapChar::None) => {
+            match keycode {
```

Comment:
Could you move `match` into `=>`  with an extra bracket?

Reply:
without the extra bracket I guess you mean - can do that

Reply:
yes, without.